tests/unit/package_managers/test_npm.py

```diff
@@ -667,6 +762,7 @@ def test_get_purl_integrity_handling(
                     "version": "1.0.0",
                     "bundled": True,
                     "dev": False,
+                    "missing_hash_in_file": [],
```

Comment:
It would also be good to test missing hashes here (the _generate_component_list function)